### PR TITLE
Add gltf sector cache

### PR DESCRIPTION
**Pull request by haakonflatval-cognite**

### CODE REVIEW

**Comment by larsmoa** [comment]
Any stats on how well it works?

**Comment by haakonflatval-cognite** [comment]
> Any stats on how well it works?

It's hard to measure objectively. The current default cache size (300) lets the user turn around at a single spot in the middle of Johan Castberg almost without loading new sectors. Do you have suggestions for better metrics?

File: viewer/core/src/datamodels/cad/CadManager.ts [hunk 1]
```diff
@@ -39,6 +39,14 @@ export class CadManager {
 
   set budget(budget: CadModelBudget) {
     this._cadModelUpdateHandler.budget = budget;
+
+    // 15_000_000 is the default rendering budget on desktop.
+    // This gives cache size of 200 on desktop on default budget
+    const REPOSITORY_CACHE_SIZE_TO_BUDGET_RATIO = 200 / 15_000_000;
```
Comment (larsmoa) [inline viewer/core/src/datamodels/cad/CadManager.ts:45]: Don't hard-code the default budget - reference `defaultCadModelBudget`

File: viewer/packages/sector-loader/src/GltfSectorLoader.ts [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+      return {
+        levelOfDetail: sector.levelOfDetail,
+        group: group,
+        instancedMeshes: [],
+        metadata: metadata,
+        modelIdentifier: sector.modelIdentifier,
+        geometryBatchingQueue: geometryBatchingQueue
+      };
+    } catch (error) {
+      MetricsLogger.trackError(error as Error, { moduleName: 'GltfSectorRepository', methodName: 'loadSector' });
```
Comment (larsmoa) [inline viewer/packages/sector-loader/src/GltfSectorLoader.ts:97]:
```suggestion
      MetricsLogger.trackError(error as Error, { moduleName: 'GltfSectorLoader', methodName: 'loadSector' });
```

**Review by larsmoa — CHANGES_REQUESTED** [review]
See comment about referencing the default budget.

**Review by larsmoa — APPROVED** [review]

**Pull request merged**